### ISSUE TITLE
datetime doesn't properly preserve zone

### ISSUE THREAD

**Issue by mockdeep** [madeintandem/jsonb_accessor#153]
When we configure `Time.zone` in our application to something different than UTC, `jsonb_accessor` does not properly deserialize the timestamp. For example:

```rb
Time.zone = "America/Los_Angeles"
Time.zone.now # => Tue, 23 Aug 2022 15:20:14.000000000 PDT -07:00
user = User.new(onboarded_at: Time.zone.now) 
user.onboarded_at # => Tue, 23 Aug 2022 15:20:14.000000000 PDT -07:00
user.save!
user.onboarded_at # => Tue, 23 Aug 2022 15:20:14.000000000 PDT -07:00
user.reload
user.onboarded_at # => Tue, 23 Aug 2022 22:20:14.000000000 PDT -07:00
                                       ^^^^^^^^^^^^^^^^^^^^^^^^ shifted by 7 hours after a reload
```

**Comment by mockdeep** [comment]
I notice in the code you respect the configured `ActiveRecord::Base.default_timezone`, but we're not looking to change the stored zone from UTC, only the in-memory zone the user sees.

**Comment by haffla** [comment]
Sorry for the late reply @mockdeep. I just tested this myself and I cannot replicate this. Could you please show the `jsonb_accessor` part of your `User` model? Also what is your `ActiveRecord::Base.default_timezone`? Is it `:utc` or `:local`?

**Comment by haffla** [comment]
@mockdeep I have just released version 1.3.6 which fixes the issue.

**Comment by mockdeep** [comment]
@haffla great, thank you!
